### PR TITLE
chore(CI): use new release workflow

### PR DESCRIPTION
**Pull request by chenjiahan**
## Summary

The new release workflow is the same as Rsbuild and Rspress, see https://github.com/web-infra-dev/rsbuild/blob/main/.github/workflows/release.yml.

`GITHUB_TOKEN` is no longer required in this workflow.

`bench-pr.yml` is deleted. We can add it back if it still need to be used.

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: CONTRIBUTING.md
Comment (Timeless0911) [inline CONTRIBUTING.md:130]: I wrote a script in Rslib that contains step 1 and 2

Comment (chenjiahan) [reply CONTRIBUTING.md:130]: nice

Comment (chenjiahan) [reply CONTRIBUTING.md]: done

**Review by fi3ework — COMMENTED** [review]
Based on https://github.com/web-infra-dev/rsbuild/commit/058993dd8bd94371e2e04cd3078a2dd37645f4b7, is there a need to restrict the "latest" dist-tag to only trigger on the main branch? This would prevent packages from being published before a PR is merged, meaning that publishing an npm package would require review by others to prevent accidental publishing.

**Review by Timeless0911 — APPROVED** [review]

**Comment by fi3ework** [comment]
> Based on [web-infra-dev/rsbuild@058993d](https://github.com/web-infra-dev/rsbuild/commit/058993dd8bd94371e2e04cd3078a2dd37645f4b7), is there a need to restrict the "latest" dist-tag to only trigger on the main branch? This would prevent packages from being published before a PR is merged, meaning that publishing an npm package would require review by others to prevent accidental publishing.

Because other PRs may be merged after the merge and before the release, we cannot guarantee that the commit being released is the final release commit. We can adjust our approach in the future based on the situation.

**Pull request merged**
